Convert between Three.js's euler rotation and Openscad's camera

Irev-Dev opened the pull request:
This felt like it was so much harder than it needed to be. Finally figured it out. The main thing I was struggling with was that I needed to change the order that the euler rotations were applied to match what openscad does (which is just sorta implicit in openscad I believe), but kinda went down a deep rabbit hole of learning about quaternions, then coming back out to realise there was a simple solution there. Feel like I have a bit more concepts under by belt now though.

https://user-images.githubusercontent.com/29681384/110857832-6bb88380-830d-11eb-9e14-c6024aa68899.mov

I managed to get the position to work as well (needs to be scaled by 10 exactly? not sure where that magic number comes from). Perspective seems off between the two cameras, more work to do I guess is what I'm saying.

Resolves #235

franknoirot commented on web/src/components/IdeViewer/IdeViewer.js:
```diff
@@ -126,7 +126,10 @@ const IdeViewer = () => {
+            <Box position={[0, 0, 0]} size={10} color="orange" />
+            <Box position={[0, 5, 0]} size={1} color="cyan" />
+            <Box position={[0, 0, 5]} size={1} color="magenta" />
+            <Box position={[5, 0, 0]} size={1} color="hotpink" />
```
Love this addition of markers.

franknoirot reviewed (approved):
This is working great! A few thoughts:

- I don't think you need that explanatory paragraph up top after this resolution right?
- Trying it out I actually enjoyed having the image remain slightly visible while dragging the cube, maybe 20-30%, because I could better gauge how much I was changing the view. What are your thoughts?
- I think we should update the default OpenSCAD camera state to match your ThreeJS camera, because as of now the initial render doesn't align to the view cube but it starts to after your first adjustment with the cube.

Irev-Dev commented:
> * I don't think you need that explanatory paragraph up top after this resolution right?

Yup gone.
> * Trying it out I actually enjoyed having the image remain slightly visible while dragging the cube, maybe 20-30%, 
because I could better gauge how much I was changing the view. What are your thoughts?

I wasn't too keen on this, but I added it and yeah, I like it. 🥇 
> * I think we should update the default OpenSCAD camera state to match your ThreeJS camera, because as of now the 
initial render doesn't align to the view cube but it starts to after your first adjustment with the cube.

Yes, so this is whole topic that I might see if openscad folks have any insight into, the main thing that made it look bad was the fov is different between the two. so I've set the three.js fov to match openscad default fov and it looks good (though in the latest openscad it's possible to change the fov in code with which might cause headaches). One thing I can't get working is when you "move the shape" with right click, they axis and the shape no long line up. This seems like a simple matter of scale. I experimented with a number of things, the thing I had most luck with was changing the openscad camera distance to something very low ([this line](https://github.com/Irev-Dev/cadhub/blob/main/api/src/docker/openscad/runScad.js#L19) ) `--camera=${px},${py},${pz},${rx},${ry},${rz},300` the last number currently 300, if I change that something very low, like 5 or less, than the moving situation becomes much better (not perfect but better), but making that very low also seems to change the (not sure of the right term) clipping distance, as in the distance at which the objects are out of sight of the camera, so it's not practical to have it that low.

There might not be a solution since we're really pushing openscad outside of its normal use. But maybe there's something we can do the three.js side too, some scaling factor?

Other things I've done.

- Changed the reference cube to a set of axes, What do you think?
- Added a loading animation
- Firing a new image load is debounced/deffered by 400 ms, makes scroll/zoom not fire off one million requests
- general polish. I added 500ms transitions to the opacity changes, too slow


https://user-images.githubusercontent.com/29681384/110915333-9a654700-836b-11eb-83e5-29cdd7bb83df.mov



With the polish I've been doing I'm a bit concerned I'm designing in code, without having thought about what it should look like prior to. It might be a good idea to take a break and give that a proper thing (not that it can't be changed later). Maybe I should be more concerned with getting a MVP working 🤷 

One simple design choice is what color scheme do we want for the openscad output? lol

Tomorrow night
![image](https://user-images.githubusercontent.com/29681384/110912869-8d932400-8368-11eb-920c-81f8d945432c.png)

Tomorrow 
![image](https://user-images.githubusercontent.com/29681384/110912853-879d4300-8368-11eb-8fd7-b99f94df3dda.png)

Solarized
![image](https://user-images.githubusercontent.com/29681384/110912837-81a76200-8368-11eb-958f-98323b8e6bf6.png)

DeepOcean
![image](https://user-images.githubusercontent.com/29681384/110912816-7b18ea80-8368-11eb-961e-159579280047.png)

Nature
![image](https://user-images.githubusercontent.com/29681384/110912798-75bba000-8368-11eb-9159-acf599c8a685.png)


BeforeDawn
![image](https://user-images.githubusercontent.com/29681384/110912775-6fc5bf00-8368-11eb-8593-c1688f77669b.png)


Starnight
![image](https://user-images.githubusercontent.com/29681384/110912758-6b010b00-8368-11eb-9b7e-b732a64f5c1c.png)

Sunset
![image](https://user-images.githubusercontent.com/29681384/110912221-caaae680-8367-11eb-9096-bdf00adf0f99.png)

Metalic
![image](https://user-images.githubusercontent.com/29681384/110912207-c41c6f00-8367-11eb-9dd0-8b845a861365.png)

Cornfield
![image](https://user-images.githubusercontent.com/29681384/110912188-be268e00-8367-11eb-8fb9-aaeabf628328.png)
